protocol/communities_messenger_test.go

```diff
@@ -4278,3 +4280,79 @@ func (s *MessengerCommunitiesSuite) TestIsDisplayNameDupeOfCommunityMember() {
 	s.Require().NoError(err)
 	s.Require().False(result)
 }
+
+func (s *MessengerCommunitiesSuite) sendImageToCommunity(sender *Messenger, chatID string) {
+	ctx := context.Background()
+	messageToSend := common.NewMessage()
+	messageToSend.ChatId = chatID
+	messageToSend.ContentType = protobuf.ChatMessage_IMAGE
+
+	// base64 image
+	var encodedB64Image string = "iVBORw0KGgoAAAANSUhEUgAAAAgAAAAIAQMAAAD+wSzIAAAABlBMVEX///+/v7+jQ3Y5AAAADklEQVQI12P4AIX8EAgALgAD/aNpbtEAAAAASUVORK5CYII"
+	decodedBytes, _ := base64.StdEncoding.DecodeString(encodedB64Image)
+
+	messageToSend.Payload = &protobuf.ChatMessage_Image{
+		Image: &protobuf.ImageMessage{
+			Format:  1, // PNG
+			Payload: decodedBytes,
+		},
+	}
+
+	response, err := sender.SendChatMessage(ctx, messageToSend)
+	s.Require().NoError(err)
+	s.Require().Len(response.Messages(), 1)
+
+	receivers := []*Messenger{s.alice, s.bob, s.owner}
+	for _, receiver := range receivers {
+		if receiver == sender {
+			continue
+		}
+		_, err = WaitOnMessengerResponse(receiver, func(response *MessengerResponse) bool {
+			return len(response.Messages()) == 1
```

Comment:
It's probably safer to wait for the actual ID of the message instead of just relying on the length. Otherwise, you might receive an old message and it would make the test flaky.

Reply:
Yeah, I'll update the test

Reply:
Done ✅ Successfully completed 1000 runs

Reply:
Did you forget to push? This condition seems the same.

Reply:
PR updated

It's my misunderstanding. First I added this check

`s.Require().Equal(sentMessage.ID, response.Messages()[0].ID)`

 after the `WaitOnMessengerResponse` returned the response.

Reply:
Yeah I get it. It's better to check inside the `WaitOnMessengerResponse` that way we are sure that the response is from the right loop and no flakiness can happen